src/net.cpp

```diff
@@ -2799,8 +2799,11 @@ std::vector<CAddress> CConnman::GetAddresses(CNode& requestor, size_t max_addres
 {
     auto local_socket_bytes = requestor.addrBind.GetAddrBytes();
     uint64_t cache_id = GetDeterministicRandomizer(RANDOMIZER_ID_ADDRCACHE)
-        .Write(requestor.addr.GetNetwork())
+        .Write(requestor.ConnectedThroughNetwork())
```

Comment:
I'm not sure if this is matters a lot, but I think that aside from the intended change of inbound onion connections, this also changes the behavior for addrs for which `HasLinkedIPv4()` ist true, because `ConnectedThroughNetwork()` calls `CNetAddr::GetNetClass()` instead of `CNetAddr::GetNetwork()`.

Reply:
Good catch, I did not notice this. I don't know enough about these IPv6 addresses that contain IPv4 addresses but it does not seem to matter here. It might be that inbound connections on IPv6 that come from these special IPv6 addresses get their own addr cache but that does not seem bad to me. It is mildly confusing that we have 3 different functions that return the network type for an address.

@naumenkogs It looks like you were the one that introduced `CNetAddr::GetNetClass()`, what do you think about this?

Reply:
I also don't think this `HasLinkedIPv4` difference matters. I don't have a preference here.